### PR TITLE
New JavaServer Pages Cheat Sheet

### PR DESCRIPTION
**Pull request by rasikapohankar**
Cheat Sheet for JSP basics.

IA Page: https://duck.co/ia/view/javaserver_pages_cheat_sheet

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: share/goodie/cheat_sheets/json/javaserver-pages.json [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    "name": "JavaServer Pages",
+    "description": "Basics of JSP",
+
+    "metadata": {
+        "sourceName": "OverAPI",
+        "sourceUrl": "http://overapi.com/jsp"
+    },
+
+    "aliases": [
+        "jsp", "jsp basics"
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/javaserver-pages.json]: Lose the `jsp basics` alias.

Comment (rasikapohankar) [inline share/goodie/cheat_sheets/json/javaserver-pages.json]: @GuiltyDolphin It won't be needed?

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=JSP Cheat Sheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/javaserver_pages_cheat_sheet_mobile.png)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=JSP Cheat Sheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/javaserver_pages_cheat_sheet_mobile.png)

File: share/goodie/cheat_sheets/json/javaserver-pages.json (hunk 1 above)
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/javaserver-pages.json]: @rasikapohankar Aliases should just be used for things that are *alternative names* of the subject; fill-words should not be used (many of these may very well be handled by the controller at some point).

Comment (rasikapohankar) [inline share/goodie/cheat_sheets/json/javaserver-pages.json]: @GuiltyDolphin Done :+1:

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=JSP Cheat Sheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/javaserver_pages_cheat_sheet_mobile.png)

**Comment by GuiltyDolphin** [comment]
@moollaza @zachthompson @zekiel Can we bump the status up on this to `development`?

**Comment by rasikapohankar** [comment]
The test has failed saying `FAIL: No Instant Answer page found with ID 'javaserver_pages_cheat_sheet'`, I don't remember seeing it as failed before :confused:. The IA page has the same ID. What should I do to fix this?

**Comment by GuiltyDolphin** [comment]
@rasikapohankar It's an issue with the metadata - it hasn't been bumped up to `development`; hopefully @moollaza @zachthompson @zekiel can bump it.

**Comment by zekiel** [comment]
The status is now `development` but it's failing the build. Also updated the one trigger example bc it said, "sheat sheet" ;) 

https://beta.duckduckgo.com/?q=Javaserver+Pages+Cheat+Sheet&ia=cheatsheet

**Comment by rasikapohankar** [comment]
Looks like it passed! :smiley: What was going wrong before?

**Comment by zekiel** [comment]
LGTM! I'm not sure what the issue was, maybe something with the IA page but not sure without investigating more. If anything odd like that happens in the future, please let us know. 

Thanks for this and your continued help!

**Comment by daxtheduck** [comment]
@rasikapohankar this is now merged! It should be live on DuckDuckGo within one week (or during the next release) and we'll be sure to notify you when it happens.

After your Instant Answer goes live for the World to use, you will be able to see [traffic information](http://docs.duckduckhack.com/maintaining/tools.html#monitor-your-instant-answer-page) about your IA as well as receive feedback directly from users. 
                
Please refer to the [maintenance guide and docs](http://docs.duckduckhack.com/maintaining/guidelines.html) to keep up with potential changes to your Instant Answer.
                
Thank you again for your help!

**Comment by rasikapohankar** [comment]
@zekiel Thank you!
Is the issue with Esperanto Cheat Sheet similar? #3052 
https://travis-ci.org/duckduckgo/zeroclickinfo-goodies/builds/130277270#L509

**Comment by zekiel** [comment]
@rasikapohankar that's a weird one. Not sure if the Travis check or my check are off but I left a comment with a potential fix.

**Comment by GuiltyDolphin** [comment]
@zekiel Whoops! I forgot to merge this - thanks!

**Pull request merged**